services/user.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AbstractBaseUser
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: Optional[str] = None,
+    first_name: Optional[str] = None,
+    last_name: Optional[str] = None,
+) -> AbstractBaseUser:
+    user = get_user_model().objects.create_user(
+        username=username, password=password
+    )
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> AbstractBaseUser:
+    return get_user_model().objects.get(pk=user_id)
+
+
+def update_user(
+    user_id: int,
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+    email: Optional[str] = None,
+    first_name: Optional[str] = None,
+    last_name: Optional[str] = None,
+) -> AbstractBaseUser:
+    user = get_user_model().objects.get(id=user_id)
```

Comment:
The `get` method will raise a `DoesNotExist` exception if no user with the given ID is found. Consider handling this exception to provide a more user-friendly error message or behavior.